src/daemon/protocols/edp.c

```diff
@@ -345,6 +347,7 @@ edp_decode(struct lldpd *cfg, char *frame, int s, struct lldpd_hardware *hardwar
 			port->p_id_subtype = LLDP_PORTID_SUBTYPE_IFNAME;
 			edp_slot = PEEK_UINT16;
 			edp_port = PEEK_UINT16;
+			free(port->p_id);
```

Comment:
Fix a memory leak.
A memory leak occurs when `case EDP_TLV_INFO` is called twice.

Scenario:
1- Call occurs.
-----> Memory leak.
2- Call occurs.
-----> Memory is freed by the program.

Therefore, if there are two or more calls to `case EDP_TLV_INFO`, all of them leak memory, except the last one.
This change handles all memory leaks.